src/solvers/refinement/string_constraint.cpp

```diff
@@ -8,6 +8,25 @@
 
 #include "string_constraint.h"
 
+#include <solvers/sat/satcheck.h>
+#include <util/symbol_table.h>
+
+/// Runs a solver instance to verify whether an expression can only be
+//  non-negative.
+/// \param expr: the expression to check for negativity
+/// \return true if `expr < 0` is unsatisfiable, false otherwise
+static bool cannot_be_neg(const exprt &expr)
+{
+  satcheck_no_simplifiert sat_check;
+  symbol_tablet symbol_table;
+  namespacet ns(symbol_table);
```

Comment:
Are you sure this will always work with this empty symbol table?

Reply:
Since we're not using anything outside of the expression we're evaluating, I think it's fine. But please tell me if you see a potential flaw.